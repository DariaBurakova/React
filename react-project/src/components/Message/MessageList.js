import React from 'react';
import './Message.css'

export const MessageList=({messageList})=> {
    return (
            <div>
                {messageList.map((item) => (
                    <div className="contentMessage  px-5">
                        <div className="answerMessage  row gx-5">
                            <div key={item.id}  className="main-answer  border border-info rounded-pill  opacity-75" >{item.text} {item.author}</div>

                        </div>
                    </div>
                        ))}
            </div>
            )
};
