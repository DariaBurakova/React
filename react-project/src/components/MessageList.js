import React from 'react';
import './Message.css'

export const MessageList=(props)=> {
    return (
        <div>
        {props.messageList.map((item) => (
            <div key={item.id} className="contentMessage  px-5">
                <div className="answerMessage  row gx-5">
                    <div  className="main-answer  border border-info rounded-pill  opacity-75" >{item.text} {item.author}</div>
                </div>
            </div>
        ))}
    </div>
    );
};


