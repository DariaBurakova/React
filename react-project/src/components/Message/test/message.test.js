import { render} from '@testing-library/react';
import {MessageList} from "../MessageList";
describe('Message test',()=>{
    it('renders text',()=>{
       const message= render(<MessageList messageList={ [{text:'text'}]}/> )
       const msgText=message.getByText('text')
        expect(msgText).toBeInTheDocument()


    })
    it("renders author", () => {
        const author= render(<MessageList messageList={ [{author:'user'}]}/> )
        const authorText = author.getByText('user');
        expect(authorText).toBeInTheDocument();
    });
    it("matches snapshot with author", () => {
        const message= render(<MessageList messageList={ [{text:'text'}]}/> )

        expect(message).toMatchSnapshot();
    });
    it("matches snapshot with no author", () => {
        const author= render(<MessageList messageList={ [{author:'user'}]}/> )

        expect(author).toMatchSnapshot();
    });
});

