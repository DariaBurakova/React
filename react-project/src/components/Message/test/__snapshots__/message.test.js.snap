// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message test matches snapshot with author 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="contentMessage  px-5"
        >
          <div
            class="answerMessage  row gx-5"
          >
            <div
              class="main-answer  border border-info rounded-pill  opacity-75"
            >
              text
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="contentMessage  px-5"
      >
        <div
          class="answerMessage  row gx-5"
        >
          <div
            class="main-answer  border border-info rounded-pill  opacity-75"
          >
            text
             
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Message test matches snapshot with no author 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="contentMessage  px-5"
        >
          <div
            class="answerMessage  row gx-5"
          >
            <div
              class="main-answer  border border-info rounded-pill  opacity-75"
            >
               
              user
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="contentMessage  px-5"
      >
        <div
          class="answerMessage  row gx-5"
        >
          <div
            class="main-answer  border border-info rounded-pill  opacity-75"
          >
             
            user
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
